### ISSUE TITLE
Outdated WorldSavedData documentation

### ISSUE THREAD

**Issue by dhharris** [MinecraftForge/Documentation#285]
**Minecraft Version:** 1.14.4

**Forge Version:** 1.14.x

**Logs:** N/A

**Steps to Reproduce:**
 1. Go to [documentation ](https://mcforge.readthedocs.io/en/1.14.x/datastorage/worldsaveddata/)for WorldSavedData
 2. Observe that the example code references methods that do not exist in MCForge

**Description of issue:**
The documentation for WorldSavedData is not just syntactically wrong, it is completely misleading. It doesn't mention the required methods `read` and `write`, and the example code (see below) references outdated methods from the Minecraft API. For example, the `World` class does not have a method `getMapStorage`. It seems like all of the data is now stored using the NBT classes in Minecraft.

```
public static ExampleWorldSavedData get(World world) {
  // The IS_GLOBAL constant is there for clarity, and should be simplified into the right branch.
  MapStorage storage = IS_GLOBAL ? world.getMapStorage() : world.getPerWorldStorage();
  ExampleWorldSavedData instance = (ExampleWorldSavedData) storage.getOrLoadData(ExampleWorldSavedData.class, DATA_NAME);

  if (instance == null) {
    instance = new ExampleWorldSavedData();
    storage.setData(DATA_NAME, instance);
  }
  return instance;
}
```
I found some good examples of WorldSavedData in the MC Forge codebase, such as the [WorldCapabilityData](https://github.com/MinecraftForge/MinecraftForge/blob/1.14.x/src/main/java/net/minecraftforge/common/util/WorldCapabilityData.java) class. For NBT serialization, [ItemStackHandler](https://github.com/MinecraftForge/MinecraftForge/blob/1.14.x/src/main/java/net/minecraftforge/items/ItemStackHandler.java) is good.

**Comment by Danjb1** [comment]
Thanks for raising this, it's helpful to have a working example to follow.

**Comment by ChampionAsh5357** [comment]
Closed by #389 and eventually #378.
